### PR TITLE
Update guestbook-logs-metrics-with-elk.md

### PR DESCRIPTION
**Pull request by Aut0R3V**
#25837
 Editing the documentation and adding in some more points to make it more focused on general skills and operational needs

### CODE REVIEW

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-25953--kubernetes-io-master-staging.netlify.app/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk/#logging

File: content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md [hunk 1]
```diff
@@ -43,6 +43,12 @@ Additionally you need:
 
 <!-- lessoncontent -->
 
+## Logging
+Generally, application logs are sent to the STDOUT and STDERR streams  and are configured by a logging driver. In the case of Kubernetes, logs are assigned via the kubelet service that runs on the node. You can have a look at the logging architecture(/docs/concepts/cluster-administration/logging) for detailed information
```
Comment (bradtopol) [inline content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md:47]: The link is not rendering properly. See the preview link provided by @kbhawkey to confirm this.  The syntax should be similar to link to something like:
[PHP Guestbook with Redis](/docs/tutorials/stateless-application/guestbook)

File: content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md [hunk 2]
```diff
@@ -43,6 +43,12 @@ Additionally you need:
 
 <!-- lessoncontent -->
 
+## Logging
+Generally, application logs are sent to the STDOUT and STDERR streams  and are configured by a logging driver. In the case of Kubernetes, logs are assigned via the kubelet service that runs on the node. You can have a look at the logging architecture(/docs/concepts/cluster-administration/logging) for detailed information 
+
+## DaemonSet
+It is required that the defined logging agent runs at every node. One of the ways to make this possible is using a DaemonSet. A DaemonSet(/docs/concepts/workloads/controllers/daemonset.md) creates a workflow that involves a copy of Pod running at each node. Hence in the cluster, as a new node is added, consecutive Pods get added.
```
Comment (bradtopol) [inline content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md:50]: The link is not rendering properly. See the preview link provided by @kbhawkey to confirm this.  The syntax should be similar to link to something like:
[PHP Guestbook with Redis](/docs/tutorials/stateless-application/guestbook)

File: content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md [hunk 3]
```diff
@@ -43,6 +43,12 @@ Additionally you need:
 
 <!-- lessoncontent -->
 
+## Logging
+Generally, application logs are sent to the STDOUT and STDERR streams  and are configured by a logging driver. In the case of Kubernetes, logs are assigned via the kubelet service that runs on the node. You can have a look at the logging architecture(/docs/concepts/cluster-administration/logging) for detailed information 
+
+## DaemonSet
```
Comment (bradtopol) [inline content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md:49]: This section seems out of place with regards to the rest of the tutorial. I did not see where DaemonSets are covered so this did not seem to be part of the flow of the tutorial.

File: content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md (hunk 1 above)
Comment (kbhawkey) [inline content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md:47]: Hi @Aut0R3V . Here is some feedback:
- Is the goal to update the tutorial by adding an overview of k8s logging and modify the example to use DaemonSets?
- Some style edits:
   - You could add a newline between a heading and the following text.
   - To link to another page, write a Markdown style link as:

```
To learn about Kubernetes basic logging, node-level logging agents, and cluster-level logging, see [logging architecture](/docs/concepts/cluster-administration/logging/).

A [DaemonSet](/docs/concepts/workloads/controllers/daemonset/) creates a ...
```

**Comment by Aut0R3V** [comment]
@bradtopol sure that sounds great. Thanks for the comments, I'll make the changes

**Comment by DanRoscigno** [comment]
@Aut0R3V I think something like this would flow easier:

# Background

Collecting operations data, specifically logs and metrics, from your Kubernetes cluster involves adoption of some best practices:
 - Streaming logs to STDOUT rather than log files
 - Autodiscovery of new resources and their logs and metrics
 - Using the environment as the configuration store

## Kubernetes facilities used to support the above

### Streaming logs to STDOUT rather than log files
why this is done, a little about the container runtime writing the streams from each pod on the node to a log file removing the responsibility of log management from the application.  Could link to https://kubernetes.io/docs/concepts/cluster-administration/logging/

### Autodiscovery of new resources and their logs and metrics
talk about the importance of integrating with the k8s api to detect that a new resource has been created or changed (e.g., change IP), determine the type of resource by looking at the k8s metadata, and start collecting the logs and metrics without human intervention.  Maybe reference the k8s REST API, specifically https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes  This is also where DaemonSets would be introduced as DaemonSets can guarantee that there is a log collector and metric collector deployed on each k8s node as those nodes are scaled / moved.

### Using the environment as the configuration store
Talk about how configuration varies between deployments and therefore the configuration should be specified in the environment specific to the deployment, and not embedded in the code.  Maybe reference https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/ and https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables

I would then move into the tutorial, introduce the PHP stateless app, and example implementation(s)

**Comment by celestehorgan** [comment]
Hi @Aut0R3V,

are you still working on this PR? It needs a rebase :)

**Comment by tengqm** [comment]
No response from author for 2 months; conflicts identified.
Please feel free to reopen and resume this when time permits.
/close

**Pull request closed without merging**